src/client.ml

```diff
@@ -0,0 +1,68 @@
+open Core_kernel
+open Async_kernel
+open Async_unix
+
+module Scheme_host_port = struct
+  module T = struct
+    type t =
+      { scheme : string
+      ; host : string
+      ; port : int
+      }
+    [@@deriving compare, hash, sexp]
+  end
+
+  include T
+  module Table = Hashtbl.Make (T)
+
+  let of_uri uri =
+    { scheme = Uri.scheme uri |> Option.value ~default:"http"
+    ; host = Uri.host_with_default ~default:"localhost" uri
+    ; port = Uri.port uri |> Option.value ~default:80
+    }
+  ;;
+end
+
+type t = Connection.t Scheme_host_port.Table.t
```

Comment:
TODO: This needs to drop old connections eventually, and also allow for concurrent connections to the same host and port